packages/flutter/test/widgets/fade_transition_test.dart

```diff
@@ -33,4 +34,25 @@ void main() {
     debugPrint = oldPrint;
     debugPrintBuildScope = false;
   });
+
+  // Regression test for https://github.com/flutter/flutter/issues/157312
+  testWidgets('No exception when calling markNeedsPaint after changing the value and then resetting the animated value to 0',
+  (WidgetTester tester) async {
```

Comment:
nit: maybe the test name could be a bit more concise?

```suggestion
  testWidgets('No exception when calling markNeedsPaint during opacity changes', (WidgetTester tester) async {
```